presidio-anonymizer/presidio_anonymizer/entities/anonymizer_result.py

```diff
@@ -0,0 +1,47 @@
+"""Handle a serializable anonymizer result."""
+import json
+
+from presidio_anonymizer.entities.anonymizer_result_item import AnonymizerResultItem
+
+
+class AnonymizerResult:
+    """Anonymizer result."""
+
+    def __init__(self, text: str = None, items: list[AnonymizerResultItem] = None):
+        """Create AnonymizerResult entity.
+
+        :param text: The anonymized text.
+        :param items: List of PII entities and their indexes in the anonymized text.
+        """
+        if items is None:
+            items = []
+        self.text = text
+        self.items = items
+
+    def set_text(self, text: str):
+        """Set a text."""
+        self.text = text
+
+    def add_item(self, item: AnonymizerResultItem):
+        """Add an item.
+
+        :param item: an item to add to the list.
+        """
+        self.items.append(item)
+
+    def normalize_item_indexes(self):
```

Comment:
Can this be done on the fly, instead of in a post process? We have an intermediate object which holds the wrong indices (0 start for example). Might help with mitigating future bugs.

Reply:
It can be done for each AnalyzerResult when it's created...

Reply:
No, because the text keep changing during the loop. When the loop is done, you can do this calculation